### PR TITLE
The final batch of dataclasses conversions

### PR DESCRIPTION
**Pull request by andlaus**
After this, all classes which correspond to XML tags in the database should finally be dataclasses.

Andreas Lauser &lt;andreas.lauser@mercedes-benz.com&gt;, on behalf of [MBition GmbH](https://mbition.io/).
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

### CODE REVIEW

File: odxtools/structure.py [hunk 1]
```diff
@@ -1,18 +1,9 @@
 # SPDX-License-Identifier: MIT
+from dataclasses import dataclass
+
 from .basicstructure import BasicStructure
 
 
+@dataclass
 class Structure(BasicStructure):
```
Comment (kayoub5) [inline odxtools/structure.py:8]: This means that `BasicStructure` and `Structure` could be merged into one class

Comment (andlaus) [reply odxtools/structure.py:8]: yes. I was just too afraid of the ripple effects...

File: odxtools/message.py [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
+from typing import TYPE_CHECKING
+
+from .odxtypes import ParameterValue, ParameterValueDict
+
+if TYPE_CHECKING:
+    from .diagservice import DiagService
+    from .structure import Structure
 
 
 class Message:
```
[the remark this answers is not in the document]
Comment (andlaus) [reply odxtools/message.py:11]: We can arguably make it one: as I see it, `Message` objects are pretty ephemeral. That said their primary purpose is to represent data. What do you prefer?

File: odxtools/structure.py (hunk 1 above)
Comment (andlaus) [reply odxtools/structure.py:8]: I got curious and looked this up in the XSD: there, `STRUCTURE` is identical to `BASIC-STRUCTURE` with an additional optional `IS-VISIBLE` attribute. Seems quite useless to me...

File: odxtools/message.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (andlaus) [reply odxtools/message.py:13]: ok. fixed: f987cd0094bfb3f8f524648f579939dd16542900

File: odxtools/message.py [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
         return self.param_dict[key]
 
-    def __str__(self):
+    def __str__(self) -> str:
         param_string = ", ".join(
             map(lambda param: f"{param[0]}={repr(param[1])}", self.param_dict.items()))
         return f"{self.structure.short_name}({param_string})"
 
-    def __repr__(self):
+    def __repr__(self) -> str:
```
[the remark this answers is not in the document]
Comment (andlaus) [reply odxtools/message.py]: right. fixed: d638f12f95a20b1b9164a0cb979469c783f7e4f2

**Review by kayoub5 — APPROVED** [review]

**Pull request merged**